core/src/main/java/com/opensymphony/xwork2/interceptor/CSRFPreventionInterceptor.java

```diff
@@ -0,0 +1,44 @@
+package com.opensymphony.xwork2.interceptor;
+
+import com.opensymphony.xwork2.ActionContext;
+import com.opensymphony.xwork2.ActionInvocation;
+
+import javax.servlet.http.HttpServletRequest;
+import java.util.ArrayList;
+import java.util.List;
+
+public class CSRFPreventionInterceptor extends AbstractInterceptor {
+
+    private List<String> exemptedPaths = new ArrayList<String>();
+    private DefaultResourceIsolationPolicy fmPolicy = new DefaultResourceIsolationPolicy();
+
+    public List<String> getExemptedPaths(){
+        return this.exemptedPaths;
+    }
+
+    public void setExemptedPaths(List<String> paths){
+        this.exemptedPaths = paths;
+    }
```

Comment:
Can't we pass these paths at build time? We should be able to have the configuration for this Interceptor right from the start, so we can set them only once
```suggestion
    public FetchMetadataInterceptor(String exemptions) {
        this.exemptedPaths = asList(exemptions);
    }
```

Reply:
I will set exempted paths only once in the constructor but I was thinking to also include setExemptedPaths, addExemptedPath and clearExemptedPaths methods for additional functionality?

Reply:
It seems unlikely that devs will want to change the set of exempted paths at runtime. Since Interceptors aren't inherently thread safe, it'd be better for us to keep our interceptor immutable!

Reply:
Got it!